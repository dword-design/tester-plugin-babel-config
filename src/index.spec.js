import { endent } from '@dword-design/functions'
import tester from '@dword-design/tester'
import packageName from 'depcheck-package-name'
import { readFile } from 'fs-extra'
import P from 'path'

import self from '.'

export default tester(
  {
    subdir: {
      cwd: 'sub',
      files: {
        'babel.config.js': endent`
          module.exports = {
            presets: [
              ['${packageName`@babel/preset-env`}', { targets: { node: 14 } }],
            ],
          }
        `,
        'sub/src/index.js': endent`
          import foo from 'foo'

          export default foo
        `,
      },
      async test() {
        expect(
          await readFile(P.join('sub', 'dist', 'index.js'), 'utf8')
        ).toMatchSnapshot(this)
      },
    },
    works: {
      files: {
        'babel.config.js': endent`
          module.exports = {
            presets: [
              ['${packageName`@babel/preset-env`}', { targets: { node: 14 } }],
            ],
          }
        `,
        'src/index.js': endent`
          import foo from 'foo'

          export default foo
        `,
      },
      async test() {
        expect(
          await readFile(P.join('dist', 'index.js'), 'utf8')
        ).toMatchSnapshot(this)
      },
    },
  },
  [self('babel.config.js')]
)
