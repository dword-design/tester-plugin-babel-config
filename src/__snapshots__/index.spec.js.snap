// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index subdir 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
var _foo = _interopRequireDefault(require(\\"foo\\"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _default = _foo.default;
exports.default = _default;"
`;

exports[`index works 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
var _foo = _interopRequireDefault(require(\\"foo\\"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _default = _foo.default;
exports.default = _default;"
`;
